### PR TITLE
feat(carne): Add carne asada parser.

### PR DESCRIPTION
**Pull request by bomt1me**
Add Carne Asada parser. Some things that I do (only for fun):
1. Thread parsing based on specified "memory limit bytes"
2. Write files by chunk and merge at the end to keep memory limit usage small

### CODE REVIEW

File: src/recipe/carne_asade.rs [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+impl CarneAsada {
+    pub fn calculate_chunks(
+        total_bytes: u32,
+        max_bytes: u32,
+        max_threads: u32,
+        step_count: u32,
+        sample_size: u32,
+    ) -> Vec<Vec<(u32, u32)>> {
+        let maxb = f64::from(max_bytes);
```
Comment (bomt1me) [inline src/recipe/carne_asade.rs]: https://en.wikipedia.org/wiki/Max_B

File: src/recipe/carne_asade.rs [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        guac: &Vec<f64>,
+    ) -> (usize, u32, String) {
+        let filename = format!("{}_{}_{}.json", onset, offset, step);
+        let file = File::create(dir.join(&filename)).expect("Could not create file.");
+        let writer = BufWriter::new(file);
+        serde_json::to_writer(writer, &guac).expect("Could not write json.");
+        (step, onset, filename)
+    }
+
+    pub fn merge(dir: &PathBuf, files: &Vec<(usize, u32, String)>) {
```
Comment (bomt1me) [inline src/recipe/carne_asade.rs]: @hwadii 😅 oh you know.. normal json stuff

Comment (hwadii) [reply src/recipe/carne_asade.rs]: haha this is unhinged ngl

Comment (bomt1me) [reply src/recipe/carne_asade.rs]: it's a private api 😉 no need for concern

File: src/recipe/carne_asade.rs [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+        guac: &Vec<f64>,
+    ) -> (usize, u32, String) {
+        let filename = format!("{onset}_{offset}_{step}.json");
+        let file = File::create(dir.join(&filename)).expect("Could not create file.");
+        let writer = BufWriter::new(file);
+        serde_json::to_writer(writer, &guac).expect("Could not write json.");
+        (step, onset, filename)
+    }
+
+    pub fn merge(dir: &Path, files: &[(usize, u32, String)]) {
```
Comment (bomt1me) [inline src/recipe/carne_asade.rs:306]: need to fix formatting :/

**Pull request merged**